src/launcher_gui.js

```diff
@@ -21,89 +26,123 @@ app.on('second-instance', () => {
 });
 
 app.prependListener('ready', () => {
-	const { wizard } = require('./launcher_wizard');
-
-	const display = electron.screen.getPrimaryDisplay();
-	const sWidth = display.workAreaSize.width;
-	const width = 800;
-	const height = process.platform === 'win32' ? 418 : 380 + 8;
-
-	let windowOpts = {
-		x: (sWidth - width) / 2,
-		// y: (sHeight - height) / 2,
-		y: 100,
-		width: width,
-		height: height,
-		show: false,
-		icon: `${__dirname}/renderer/spring-icon.png`,
-		webPreferences: {
-			nodeIntegration: true,
-			enableRemoteModule: true,
-		},
-	};
-	windowOpts.resizable = false;
-	Menu.setApplicationMenu(null);
-	mainWindow = new BrowserWindow(windowOpts);
-
-	mainWindow.loadFile(`${__dirname}/renderer/index.html`);
-	if (isDev) {
-		// mainWindow.webContents.openDevTools();
-	}
+	// TODO: is there a better way to get yargs' positional args ? Did not find
+	// it ...
+	if (argv._[0]) {
+		// If a replay file was passed as argument to spring-launcher, then we
+		// don't open the GUI, we just start the replay.
+
+		// TODO: when we support downloading assets (map/game/engine) we might
+		// need to show the GUI anyway, to provide feedback on the
+		// downloads/ask for user confirmation.
```

Comment:
even now it might be useful to have since errors aren't caught properly

i.e. I get this when I try to launch an invalid replay: 

```
TypeError: Cannot read property 'show' of undefined
    at Launcher.<anonymous> (/home/gajop/radni_direktorijum/programi/spring-launcher/src/main.js:50:13)
    at Launcher.emit (events.js:315:20)
    at ChildProcess.<anonymous> (/home/gajop/radni_direktorijum/programi/spring-launcher/src/engine_launcher.js:208:9)
    at ChildProcess.emit (events.js:315:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
TypeError: Cannot read property 'show' of undefined
    at Launcher.<anonymous> (/home/gajop/radni_direktorijum/programi/spring-launcher/src/main.js:50:13)
    at Launcher.emit (events.js:315:20)
    at ChildProcess.<anonymous> (/home/gajop/radni_direktorijum/programi/spring-launcher/src/engine_launcher.js:208:9)
    at ChildProcess.emit (events.js:315:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Reply:
What is an invalid replay ? Like a file that is not a replay ? I'm going to try and reproduce

Reply:
Ok I figured it out. Basically anything failing at the engine level would be handled wrongly by the `'failed'` handler in `main.js`. This is now fixed.